src/python/pants/engine/target.py

```diff
@@ -1705,12 +1720,25 @@ def compute_value(
 class StringSequenceField(SequenceField[str]):
     expected_element_type = str
     expected_type_description = "an iterable of strings (e.g. a list of strings)"
+    valid_choices: ClassVar[Optional[Union[Type[Enum], Tuple[str, ...]]]] = None
 
     @classmethod
     def compute_value(
         cls, raw_value: Optional[Iterable[str]], address: Address
     ) -> Optional[Tuple[str, ...]]:
-        return super().compute_value(raw_value, address)
+        value_or_default = super().compute_value(raw_value, address)
+        if value_or_default and cls.valid_choices is not None:
+            valid_choices = set(
+                cls.valid_choices
+                if isinstance(cls.valid_choices, tuple)
+                else (choice.value for choice in cls.valid_choices)
+            )
+            for choice in value_or_default:
```

Comment:
Can we unify this with the existing implementation for StringField? Maybe refactor that out and use it in both places? 

To streamline that I think it would be fine to raise InvalidFieldChoiceException in the sequence field case as well, rather than create the new exception type. The language of existing message can be tweaked to make sense for both cases, e.g., "Values for the ... field in target ... must be one of ..., but ... was provided".

Reply:
Ah cool. I used `InvalidFieldChoiceException` initially but tried to clarify the message. Your suggestion makes it work for both, so I dropped `InvalidSequenceFieldChoiceException`.

As far as unifying the choices validation logic, are you suggesting I create a function and use it in both places? Or some kind of choice validation mixin class?

Reply:
Maybe just a helper function? It can operate on a list of values, and we can pass in a singleton list for the StringField case?

Reply:
K. I refactored this into a helper function.